src/oscar/core/application.py

```diff
@@ -1,8 +1,53 @@
-from django.apps import AppConfig
+from django.apps import AppConfig, apps
 from django.core.exceptions import ImproperlyConfigured
-from django.urls import URLPattern, reverse_lazy
+from django.urls import URLPattern, include, path, re_path, reverse_lazy
 
-from oscar.core.loading import feature_hidden
+
+class AutoLoadURLsConfigMixin:
+    include_urls_in_parent = True
```

Comment:
Previously, apps in `OSCAR_HIDDEN_FEATURES` had their URLs [excluded](https://github.com/django-oscar/django-oscar/blob/master/src/oscar/core/application.py#L82) (the referenced issue #3414 claims otherwise, or am I misunderstanding something?), in addition to having template sections hidden using the `iffeature` template tag. To achieve the same, one should now remove the app config from `INSTALLED_APPS`.

With this attribute (which should be in the child app config class), we are adding the option for one to exclude the URLs of installed apps. What is the use case for this? And why are we doing this only for child apps, and not all apps in general?

If one really wants to do this (and is ready to deal with the resulting `django.urls.NoReverseMatch` errors), how about they just override the app's `OscarConfig.get_urls()` to return an empty list?

@solarissmoke?

Reply:
@jwayodi Sorry,`include_urls_in_parent` name is/was a bit misleading. It's meant to create a differentiate between two usecases `path("endpoint/", include(...))` and `path("endpoint/", ...)`.